src/sentry/api/endpoints/organization_member_invite/index.py

```diff
@@ -0,0 +1,337 @@
+            )
+
+        return email
+
+    def validate_orgRole(self, role):
+        if role == "billing" and features.has(
+            "organizations:invite-billing", self.context["organization"]
+        ):
+            return role
+        role_obj = next((r for r in self.context["allowed_roles"] if r.id == role), None)
+        if role_obj is None:
+            raise serializers.ValidationError(
+                "You do not have permission to invite a member with that org-level role"
+            )
+        if not self.context.get("allow_retired_roles", True) and role_obj.is_retired:
+            raise serializers.ValidationError(
+                f"The role '{role}' is deprecated, and members may no longer be invited with it."
+            )
+        return role
+
+    def validate_teams(self, teams):
+        valid_teams = list(
+            Team.objects.filter(
+                organization=self.context["organization"], status=TeamStatus.ACTIVE, slug__in=teams
+            )
+        )
+
+        if len(valid_teams) != len(teams):
+            raise serializers.ValidationError("Invalid teams")
+
+        return valid_teams
+
+
+@region_silo_endpoint
+@extend_schema(tags=["Organizations"])
+class OrganizationMemberInviteIndexEndpoint(OrganizationEndpoint):
+    # TODO (mifu67): make these PUBLIC once ready
+    publish_status = {
+        "GET": ApiPublishStatus.EXPERIMENTAL,
+        "POST": ApiPublishStatus.EXPERIMENTAL,
+    }
+    permission_classes = (MemberInviteAndStaffPermission,)
+    owner = ApiOwner.ENTERPRISE
+
+    def _can_invite_member(
+        self,
+        request: Request,
+        organization: Organization | RpcOrganization | RpcUserOrganizationContext,
+    ) -> bool:
+        scopes = request.access.scopes
+        is_role_above_member = "member:admin" in scopes or "member:write" in scopes
+        if isinstance(organization, RpcUserOrganizationContext):
+            organization = organization.organization
+
+        if is_role_above_member:
+            return True
+        if "member:invite" not in scopes:
+            return False
+        return not organization.flags.disable_member_invite
+
+    def _invite_member(self, request, organization) -> Response:
+        assigned_org_role = request.data.get("orgRole") or organization_roles.get_default().id
+        allowed_roles = get_allowed_org_roles(request, organization, creating_org_invite=True)
+
+        # We allow requests from integration tokens to invite new members as the member role only
+        if not allowed_roles and request.access.is_integration_token:
+            # Error if the assigned role is not a member
+            if assigned_org_role != "member":
+                raise serializers.ValidationError(
+                    "Integration tokens are restricted to inviting new members with the member role only."
+                )
+            allowed_roles = [organization_roles.get("member")]
```

Comment:
you might be able to put this check in the serializer and pass `request.access.is_integration_token` in `context`, since this is raising `serializers.ValidationError`. that way you don't have to explicitly fetch the `orgRole` and check this outside of the serializer, it will be covered in `validate_orgRole`